Adding cypress tests to check dataflow-table linking and page reload …

tejal-shah opened the pull request:
…checks for various states of play/stop/record/pause buttons

scytacki reviewed (changes requested):
This looks good, but the tests depend on each other.
I was thinking that we would stop adding new tests like that.

However I've never been happy with the options for making independent tests, so I don't know what to suggest.

You could put all of the dependent code into a single test, but then there isn't a nice way to divide and describe different features.

You could add setup code for each of the tests, but this will slow things down.

Something new you could consider is using the `/doc-editor.html` for this kind of testing. With that you could load in pre configured documents for each test. That way you wouldn't have to use UI to setup the document. I don't think we've tried to drive the file open dialog of `/doc-editor.html` with cypress though, so it might be hard to use it that way.  It wouldn't take much work to add a url param specifying a file for the doc-editor to load, so if this seems like something you want to try, I can squeeze in the time to add that new parameter.

tejal-shah commented:
@scytacki , I've made a few improvements in my latest commit in this branch:
1. I've moved the `dataflow-table linking-button states upon page reload` tests into a separate spec file so as to keep the linking tests separate (this follows the pattern of graph-table linked tests)
2. I've moved the document setup into a `beforeEach` method to keep the tests independent.
3. The tests themselves are longer now than before, but I've tried to keep the code more understandable by moving the granular verifications into the support files so that the spec files show a big picture of what's being tested in each test.

Please take a look and let me know what you think.

scytacki reviewed (approved):
looks great to me 👍
